apm-commons/apm-util/src/main/java/org/apache/skywalking/apm/util/ConfigInitializer.java

```diff
@@ -50,43 +53,45 @@ private static void initNextLevel(Properties properties, Class<?> recentConfigTy
                 String configKey = (parentDesc + "." + field.getName()).toLowerCase();
                 Class<?> type = field.getType();
 
-                if (type.equals(Map.class)) {
+                if (Map.class.isAssignableFrom(type)) {
                     /*
-                     * Map config format is, config_key[map_key]=map_value
-                     * Such as plugin.opgroup.resttemplate.rule[abc]=/url/path
+                     * Map config format is, config_key[map_key]=map_value, such as plugin.opgroup.resttemplate.rule[abc]=/url/path
+                     * "config_key[]=" will generate a bleak Map , user could use this mechanism to set a blank Map
                      */
-                    // Deduct two generic types of the map
                     ParameterizedType genericType = (ParameterizedType) field.getGenericType();
                     Type[] argumentTypes = genericType.getActualTypeArguments();
-
-                    Type keyType = null;
-                    Type valueType = null;
-                    if (argumentTypes != null && argumentTypes.length == 2) {
-                        // Get key type and value type of the map
-                        keyType = argumentTypes[0];
-                        valueType = argumentTypes[1];
+                    Type keyType = argumentTypes[0];
+                    Type valueType = argumentTypes[1];
+                    // A chance to set a blank map
+                    if (properties.containsKey(configKey + "[]")) {
+                        field.set(null, initBlankMap(type));
```

Comment:
If it has been initialized as empty, you should break and jump out of the loop. Otherwise, you would provide a sequential re-init